File: week6/pset6/readability/readability.py

def main():
    text = get_text()

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    L = letters/words * 100
    S = sentences/words * 100
    index = round(0.0588 * L - 0.296 * S - 15.8)

    if index > 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")


def count_letters(text):
    counter = 0
    for i in text:
        if i.isalpha():
            counter += 1

    return counter


def count_words(text):
    counter = 1
    for i in text:
        if i == ' ':
            counter += 1

    return counter


def count_sentences(text):
    counter = 0
    for i in text:
        if i == '!' or i == '?' or i == '.':
            counter += 1

    return counter


def get_text():
    return input("Text: ")


if __name__ == "__main__":
    main()

File: week2/pset2/readability/readability.c
#include <stdio.h>
#include <string.h>
#include <cs50.h>
#include <math.h>

int numberOfLetters(string text);
int numberOfWords(string text);
int numberOfSentences(string text);
void ColemanLiauIndex(string text);

int main(void)
{
    string text = get_string("Text: ");
    ColemanLiauIndex(text);
}

int numberOfWords(string text)
{
    int counter = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == ' ')
        {
            counter++;
        }
    }
    return ++counter;
}

int numberOfSentences(string text)
{
    int counter = 0;
    for (int j = 0, n = strlen(text); j < n; j++)
    {
        if (text[j] == '.' || text[j] == '!' || text[j] == '?')
        {
            counter++;
        }
    }
    return counter;
}

int numberOfLetters(string text)
{
    int counter = 0;
    for (int k = 0, n = strlen(text); k < n; k++)
    {
        int code = text[k];
        if ((65 <= code && code  <= 90)
            || (97 <= code && code <= 122))
        {
            counter++;
        }
    }
    return counter;
}

void ColemanLiauIndex(string text)
{
    int letters = numberOfLetters(text);
    int words = numberOfWords(text);
    int sentences = numberOfSentences(text);

    int index = round((0.0588 * 100 * letters / words) - (0.296 * 100 * sentences / words) - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}